android/app/src/main/java/io/mosip/registration_client/api_services/MachineDetailsApi.java

```diff
@@ -4,54 +4,76 @@
 
 import androidx.annotation.NonNull;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
+import io.mosip.registration.clientmanager.entity.RegistrationCenter;
+import io.mosip.registration.clientmanager.repository.RegistrationCenterRepository;
 import io.mosip.registration.keymanager.spi.ClientCryptoManagerService;
 import io.mosip.registration_client.model.MachinePigeon;
 
 
 @Singleton
 public class MachineDetailsApi  implements MachinePigeon.MachineApi {
-    Map<String, String> machineDetails = new HashMap<>();
-    MachinePigeon.Machine machine;
     ClientCryptoManagerService clientCryptoManagerService;
+    RegistrationCenterRepository registrationCenterRepository;
 
     @Inject
-    public MachineDetailsApi(ClientCryptoManagerService clientCryptoManagerService) {
+    public MachineDetailsApi(ClientCryptoManagerService clientCryptoManagerService,
+                             RegistrationCenterRepository registrationCenterRepository) {
         this.clientCryptoManagerService = clientCryptoManagerService;
+        this.registrationCenterRepository = registrationCenterRepository;
     }
 
-    @NonNull
     @Override
-    public MachinePigeon.Machine getMachineDetails() {
+    public void getMachineDetails(@NonNull MachinePigeon.Result<MachinePigeon.Machine> result) {
+        Map<String, String> machineDetails = new HashMap<>();
         Map<String, String> details =
                 clientCryptoManagerService.getMachineDetails();
 
         if(details.get("name") == null) {
-            machine = new MachinePigeon.Machine.Builder()
+            MachinePigeon.Machine machine = new MachinePigeon.Machine.Builder()
                     .setMap(machineDetails)
                     .setErrorCode("REG_MACHINE_NOT_INITIALIZED")
                     .build();
-            return machine;
+            result.success(machine);
         }
 
         try {
             details.put("version", "Alpha");
-            machine = new MachinePigeon.Machine.Builder()
+            MachinePigeon.Machine machine = new MachinePigeon.Machine.Builder()
                     .setMap(details)
                     .build();
+            result.success(machine);
         } catch (Exception e) {
             Log.e(getClass().getSimpleName(), e.getMessage(), e);
-            machine = new MachinePigeon.Machine.Builder()
+            MachinePigeon.Machine machine = new MachinePigeon.Machine.Builder()
                     .setMap(machineDetails)
                     .setErrorCode("REG_MACHINE_NOT_FETCHED")
                     .build();
+            result.success(machine);
         }
+    }
 
-        return machine;
+    @Override
+    public void getCenterName(@NonNull String regCenterId, @NonNull MachinePigeon.Result<String> result) {
```

Comment:
If this is needed to display the center name on the home page then this should be based on the login language. so you should be getting the center name for a language code.

Reply:
There is only function to add center name based on id. Need to work on this and add the required functions

Reply:
ok lets create one task and add each new function as subtask